src/test/java/com/expediagroup/hiveberg/TestPredicatePushdown.java

```diff
@@ -0,0 +1,175 @@
+import java.io.IOException;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hive.ql.exec.SerializationUtilities;
+import org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat;
+import org.apache.hadoop.hive.ql.io.sarg.ConvertAstToSearchArg;
+import org.apache.hadoop.hive.ql.io.sarg.SearchArgument;
+import org.apache.hadoop.hive.ql.metadata.DefaultStorageHandler;
+import org.apache.hadoop.hive.ql.metadata.HiveStoragePredicateHandler;
+import org.apache.hadoop.hive.ql.plan.ExprNodeDesc;
+import org.apache.hadoop.hive.ql.plan.ExprNodeGenericFuncDesc;
+import org.apache.hadoop.hive.serde2.AbstractSerDe;
+import org.apache.hadoop.hive.serde2.ColumnProjectionUtils;
+import org.apache.hadoop.hive.serde2.Deserializer;
+import org.apache.hadoop.mapred.InputFormat;
+import org.apache.hadoop.mapred.InputSplit;
+import org.apache.hadoop.mapred.JobConf;
+import org.apache.hadoop.mapred.OutputFormat;
+import org.apache.iceberg.CombinedScanTask;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.FileFormat;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.data.Record;
+import org.apache.iceberg.expressions.Expression;
+import org.apache.iceberg.hadoop.HadoopCatalog;
+import org.apache.iceberg.types.Types;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+
+import static com.expediagroup.hiveberg.TableResolverUtil.pathAsURI;
+import static com.expediagroup.hiveberg.TableResolverUtil.resolveTableFromJob;
+import static org.apache.iceberg.types.Types.NestedField.optional;
+import static org.apache.iceberg.types.Types.NestedField.required;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(StandaloneHiveRunner.class)
+public class TestPredicatePushdown {
+
+  @HiveSQL(files = {}, autoStart = true)
+  private HiveShell shell;
+
+  @Rule
+  public TemporaryFolder temp = new TemporaryFolder();
+
+  private File tableLocation;
+  private Table table;
+
+  @Before
+  public void before() throws IOException {
+    tableLocation = temp.newFolder();
+    Schema schema = new Schema(required(1, "id", Types.LongType.get()),
+        optional(2, "data", Types.StringType.get()));
+    PartitionSpec spec = PartitionSpec.unpartitioned();
+
+    Configuration conf = new Configuration();
+    HadoopCatalog catalog = new HadoopCatalog(conf, tableLocation.getAbsolutePath());
+    TableIdentifier id = TableIdentifier.parse("source_db.table_a");
+    table = catalog.createTable(id, schema, spec);
+  }
+
+  /**
+   * This test is to recreate an old bug found in the StorageHandler where the filter property wasn't
+   * getting reset between queries.
+   */
+  @Test
+  public void oldVersionOfStorageHandler() throws IOException {
```

Comment:
People in the future won't know there was an old version of the storage handler. So I don't think we need the old handler and old IF here. Can you just have the test the other way around checking the new behaviour with an explanation that it's testing the filtering etc. That way we have it to prevent future regressions.

Reply:
Definitely a more robust way to do it, change has been made :)